M2/Macaulay2/m2/monoids.m2

```diff
@@ -474,14 +474,22 @@ checkSymbol = sym -> if instance(sym, Symbol) or lookup(symbol <-, class sym) =!
 -- turns {x, y, z, y} into {x, y_0, z, y_1}
 -- adding 'toString' in a few places will eliminate more duplications
 -- but makes creating temporary rings in functions more difficult.
-dedupSymbols = varlist -> if 0 == repeats varlist then varlist else while 0 < repeats varlist do (
-    mapping := hashTable toList pairs varlist;
-    -- TODO: applyPairs with collision handler, and accepting MutableHashTable
-    counter := new MutableHashTable from applyKeys(mapping,
-	i -> varlist#i, dups -> makeVars(#dups, first dups));
```

Comment:
This could have been fixed with a much simpler patch:
```suggestion
    counter := applyKeys(mapping, i -> varlist#i, splice);
    counter  = applyValues(counter, dups -> makeVars(#dups, first dups));
    counter  = new MutableHashTable from counter;
```